TYPO3.Neos/Resources/Public/JavaScript/Content/Components/ContentContextBar.js

```diff
@@ -2,38 +2,55 @@
  * Context context bar
  */
 define(
-[
-	'emberjs',
-	'./ContextBar',
-	'./ContentDimensionSelector',
-	'../FullScreenController',
-	'../Application',
-	'Shared/Configuration',
-	'text!./ContentContextBar.html'
-], function(
-	Ember,
-	ContextBar,
-	ContentDimensionSelector,
-	FullScreenController,
-	ContentModule,
-	Configuration,
-	template
-) {
-	return ContextBar.extend({
-		classNames: 'neos-content-context-bar',
-		template: Ember.Handlebars.compile(template),
-		ContentDimensionSelector: ContentDimensionSelector,
-		fullScreenController: FullScreenController,
-		Configuration: Configuration,
-		init: function() {
-			this.updateCurrentUri();
-			var that = this;
-			ContentModule.on('pageLoaded', function() {
-				that.updateCurrentUri();
-			});
-		},
-		updateCurrentUri: function() {
-			this.set('liveUri', location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, ''));
-		}
-	});
-});
\ No newline at end of file
+  [
+    'emberjs',
+    './ContextBar',
+    './ContentDimensionSelector',
+    './ContentDimensionController',
+    '../FullScreenController',
+    '../Application',
+    './TargetWorkspaceController',
+    'Shared/Configuration',
+    'text!./ContentContextBar.html'
+  ], function (Ember,
+               ContextBar,
+               ContentDimensionSelector,
+               ContentDimensionController,
+               FullScreenController,
+               ContentModule,
+               TargetWorkspaceController,
+               Configuration,
+               template) {
+    return ContextBar.extend({
+      classNames: 'neos-content-context-bar',
+      template: Ember.Handlebars.compile(template),
+      ContentDimensionSelector: ContentDimensionSelector,
+      contentDimensionController: ContentDimensionController,
+      fullScreenController: FullScreenController,
+      targetWorkspaceController: TargetWorkspaceController,
+      Configuration: Configuration,
+
+      /**
+       * Update the preview uri when the page is loaded (happens when navigating, switching
+       * target workspace or changing content dimensions).
+       */
+      init: function () {
+        var that = this;
+        ContentModule.on('pageLoaded', function () {
+          that.updatePreviewUri();
+        });
+      },
+
+      /**
+       * Update the preview uri
+       */
+      updatePreviewUri: function () {
+        var targetWorkspaceName = this.get('targetWorkspaceController.targetWorkspace.name');
+        if (targetWorkspaceName === 'live') {
+          this.set('previewUri', location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, ''));
+        } else {
+          this.set('previewUri', location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, '@' + targetWorkspaceName));
+        }
+      }.observes('targetWorkspaceController.targetWorkspace', 'contentDimensionController.selectedDimensions')
```

Comment:
it works as expected, however I'm not sure if the last change is even needed.. since the ``pageLoaded`` event is fired when you change the target workspace cause the page is reloaded.. it might cause it to update immediately instead of waiting for the page to be loaded again, but maybe keep it as is since that page reload might actually not be desired and removed at some point..

also why listen to selectedDimensions when it has not influence on the url?